keps/sig-node/4009-add-cdi-devices-to-device-plugin-api/kep.yaml

```diff
@@ -0,0 +1,44 @@
+title: Add CDI devices to device plugin API
+kep-number: 4009
+authors:
+  - "@elezar"
+owning-sig: sig-node
+participating-sigs: []
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+creation-date: 2023-05-16
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+see-also:
+  - "/keps/sig-node/3063-dynamic-resource-allocation"
+  - "/keps/sig-node/3573-device-plugin"
+replaces: []
+
+# The target maturity stage in the current dev cycle for this KEP.
+stage: alpha|beta|stable
+
+# The most recent milestone for which work toward delivery of this KEP has been
+# done. This can be the current (upcoming) milestone, if it is being actively
+# worked on.
+latest-milestone: "v1.28"
+
+# The milestone at which this feature was, or is targeted to be, at each stage.
+milestone:
+  alpha: "v1.28"
+  beta: "v1.29"
+  stable: "v1.30"
+
+# The following PRR answers are required at alpha release
+# List the feature gate name and the components for which it must be enabled
+feature-gates:
+  - name: MyFeature
```

Comment:
We need to specify the feature gate here. Adding a comment so we don't forget.

Reply:
Looking at the changes made for the addition of the CDIDevices fields to the CRI spec itself, this doesn't seem to be gated by a feature flag -- although the usage is gated by the `kubefeatures.DynamicResourceAllocation` flag. Does this then required a new feature flag given that this is an extension of the existing device plugin API?

Reply:
> Looking at the changes made for the addition of the CDIDevices fields to the CRI spec itself, this doesn't seem to be gated by a feature flag -- although the usage is gated by the kubefeatures.DynamicResourceAllocation flag. 

The intent is to guard against the usage of new changes made to the API unless explicitly specified. Since DynamicResourceAllocation is still an alpha feature it makes sense to not have a new feature gate for additional API changes as it pertains to the feature evolving while it is in alpha stage. 

> Does this then required a new feature flag given that this is an extension of the existing device plugin API?

Yes, I think this would require a new feature gate as we want new changes to be guarded by that gate and not be enabled by default just because the device plugin framework and the API is mature. This is to ensure that the newly introduced changes don't cause unintended regressions and it goes through its own graduation process maturing over multiple releases.